native/src/main/java/io/ballerina/stdlib/graphql/runtime/engine/Engine.java

```diff
@@ -178,4 +183,33 @@ private static void getExecutionResult(ExecutionContext executionContext, BObjec
                                                    strandMetadata, callback, null, PredefinedTypes.TYPE_NULL, args);
         }
     }
+
+    public static Object getSubscriptionResult(Environment env, BObject visitor, BObject node) {
+        Future subscriptionFutureResult = env.markAsync();
+        SubscriptionCallback subscriptionCallback = new SubscriptionCallback(subscriptionFutureResult);
+        CallbackHandler callbackHandler = new CallbackHandler(subscriptionFutureResult);
+        ExecutionContext executionContext = new ExecutionContext(env, visitor, callbackHandler, SUBSCRIPTION);
+        BString fieldName = node.getStringValue(NAME_FIELD);
+        BObject engine = visitor.getObjectValue(ENGINE_FIELD);
+        BObject service = (BObject) engine.getNativeData(GRAPHQL_SERVICE_OBJECT);
+        ServiceType serviceType = (ServiceType) service.getType();
+        UnionType typeUnion = TypeCreator.createUnionType(PredefinedTypes.TYPE_STREAM, PredefinedTypes.TYPE_ERROR);
+        for (ResourceMethodType resourceMethod : serviceType.getResourceMethods()) {
+            if (Objects.equals(String.valueOf(fieldName), resourceMethod.getResourcePath()[0]) &&
+                Objects.equals(resourceMethod.getAccessor(), SUBSCRIBE_ACCESSOR)) {
```

Comment:
Let's not use `Objects.equals()` method, to maintain consistency. Please check other places as well.

```suggestion
            if (SUBSCRIBE_ACCESSOR.equals(resourceMethod.getAccessor()) &&
                fieldName.getValue().equals(resourceMethod.getResourcePath()[0]) {
```